pkg/supportbundle/spec.go

```diff
@@ -251,6 +264,120 @@ func injectDefaults(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle,
 	return supportBundle, nil
 }
 
+func addDiscoveredSpecs(
+	supportBundle *troubleshootv1beta2.SupportBundle, app apptypes.AppType,
+) *troubleshootv1beta2.SupportBundle {
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		logger.Errorf("Failed to get kubernetes clientset: %v", err)
+		return supportBundle
+	}
```

Comment:
Can we pass a k8s client to this function instead of instantiating a new one?  The caller already has one, so it should be a small change.  That way we can unit test this function too.